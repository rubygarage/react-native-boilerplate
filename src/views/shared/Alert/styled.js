import styled from 'styled-components/native';

import Icon from 'views/shared/Icon';

const Wrapper = styled.View`
  position: relative;
  flex-direction: row;
  margin-bottom: 8;
  padding-top: 12;
  padding-right: ${({ onCloseClick }) => (onCloseClick ? 48 : 12)};
  padding-bottom: 12;
  padding-left: 12;
  border-radius: 4;
  background-color: ${({ color }) => color};
`;

const IconWrap = styled.View`
  margin-right: 12;
  flex-shrink: 0;
`;

const ContentArea = styled.View`
  flex-grow: 1;
  padding-right: 36;
  display: flex;
  align-items: flex-start;
`;

const MessageTitle = styled.Text`
  flex-direction: row;
  flex-wrap: wrap;
  margin-top: 2;
  font-size: 14;
  line-height: 20;
  font-weight: bold;
  color: ${({ color }) => color};
`;

const MessageText = styled.Text`
  margin-bottom: 2;
  font-size: 14;
  line-height: 20;
  color: ${({ color }) => color};
`;

const ActionButton = styled.TouchableOpacity`
  flex-direction: row;
  align-items: center;
  max-width: 200;
  height: 36;
  margin-top: 8;
  padding-right: 12;
  padding-left: 12;
  border-radius: 4;
  background-color: ${({ color }) => color};
`;

const ActionIcon = styled(Icon)`
  margin-right: 24;
`;

const ActionButtonText = styled.Text`
  color: ${({ theme }) => theme.colors.white};
  font-size: 14;
  line-height: 20;
  font-weight: bold;
  letter-spacing: -0.2;
`;

const CloseButton = styled.TouchableOpacity`
  position: absolute;
  top: 12;
  right: 12;
  color: ${({ theme }) => theme.colors.white};
`;

export {
  Wrapper,
  IconWrap,
  ContentArea,
  MessageTitle,
  MessageText,
  ActionButton,
  ActionButtonText,
  CloseButton,
  ActionIcon,
};
