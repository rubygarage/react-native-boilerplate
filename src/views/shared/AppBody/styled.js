import styled from 'styled-components/native';

const AppBody = styled.View`
  position: relative;
  flex-grow: 1;
  padding-right: 16px;
  padding-left: 16px;
  background-color: ${({ theme }) => theme.colors.black};
`;

export default AppBody;
