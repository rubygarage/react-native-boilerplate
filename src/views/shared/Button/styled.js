import styled from 'styled-components/native';

const Wrap = styled.TouchableOpacity`
  min-width: ${(props) => (props.isSmallWidth ? 72 : 136)};
  height: ${(props) => (props.isSmallHeight ? '36px' : '48px')};
  justify-content: center;
  align-items: center;
  padding-right: 8;
  padding-left: 8;
  border-radius: 4;
  flex-shrink: 0;
  background-color: ${({ theme, isOutline, fillColor }) => (isOutline ? theme.colors.white : fillColor)};
  margin-right: ${(props) => (props.isMarginRight ? '16' : '0')};
  margin-bottom: ${(props) => (props.isMarginBottom ? '16' : '0')};
  border-width: ${(props) => (props.isBorder ? '2' : '0')};
  border-style: solid;
  border-color: ${({ theme }) => theme.colors.black};
`;

const Text = styled.Text`
  color: ${({ theme, isOutline }) => (isOutline ? theme.colors.black : theme.colors.white)};
  font-weight: bold;
`;

export { Wrap, Text };
