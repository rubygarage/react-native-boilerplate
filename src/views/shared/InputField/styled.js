import styled from 'styled-components/native';

const Wrap = styled.View`
  flex-direction: column;
`;

const Label = styled.Text`
  margin-bottom: 4px;
  font-size: 14px;
  line-height: 20px;
  color: ${({ theme }) => theme.colors.black};
`;

const FieldWrap = styled.View`
  position: relative;
`;

const Field = styled.TextInput`
  border-radius: 3px;
  min-height: 48px;
  border-width: ${({ isFocused, isError }) => (isFocused && !isError ? '2px' : '1px')};
  border-style: solid;
  border-color: ${({ theme, isError, isFocused }) => {
    if (isError) {
      return theme.colors.black;
    }
    if (isFocused) {
      return theme.colors.black;
    }
    return theme.colors.black;
  }};
  padding-top: 10px;
  padding-right: ${({ actionWidth }) => (actionWidth + 12)};
  padding-bottom: 10px;
  padding-left: 12px;
  font-size: 14px;
  line-height: 20px;
  color: ${({ theme }) => theme.colors.black};
`;

const FieldError = styled.View`
  flex-direction: row;
  align-items: center;
  margin-top: 6px;
  margin-bottom: 8px;
`;

const FieldErrorClose = styled.View`
  margin-right: 8px;
`;

const FieldErrorMessage = styled.Text`
  color: ${({ theme }) => theme.colors.black};
  font-size: 14px;
  line-height: 20px;
`;

const ActionButtonWrapper = styled.View`
  position: absolute;
  height: 48px;
  right: 8px;
  justify-content: center;
`;

export {
  Wrap, Label, FieldWrap, Field,
  FieldError, FieldErrorClose, FieldErrorMessage, ActionButtonWrapper,
};
