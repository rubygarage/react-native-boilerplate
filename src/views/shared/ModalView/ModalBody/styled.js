import styled from 'styled-components/native';

const Wrap = styled.View`
  padding-top: 24px;
  padding-right: 24px;
  padding-left: 24px;
`;

// eslint-disable-next-line import/prefer-default-export
export { Wrap };
