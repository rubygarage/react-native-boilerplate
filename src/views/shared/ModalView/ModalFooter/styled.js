import styled from 'styled-components/native';

const Wrap = styled.View`
  flex-shrink: 0;
  flex-direction: row;
  justify-content: flex-end;
  padding: 16px;
`;

// eslint-disable-next-line import/prefer-default-export
export { Wrap };
