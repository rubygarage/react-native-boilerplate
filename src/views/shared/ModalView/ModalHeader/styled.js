import styled from 'styled-components/native';

const Wrap = styled.View`
  position: relative;
  flex-direction: row;
  align-items: center;
  flex-shrink: 0;
  padding: 16px 24px;
  border-bottom-color: rgba(82,97,115, 0.12);
  border-bottom-width: 1;
  border-style: solid;
`;

const Title = styled.Text`
  font-size: 16;
  line-height: 24;
  letter-spacing: -0.2;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.black};
`;

const Close = styled.TouchableOpacity`
  flex-shrink: 0;
  margin-right: 16;
`;

export { Wrap, Title, Close };
