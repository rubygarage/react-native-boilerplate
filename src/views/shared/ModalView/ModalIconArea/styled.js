import styled from 'styled-components/native';

const Wrap = styled.View`
  justify-content: center;
  align-items: center;
  margin-top: 24px;
  margin-bottom: 24px;
`;

const Icon = styled.View`
  margin-bottom: 16px;
`;

const Text = styled.Text`
  font-size: 14px;
  line-height: 20px;
  color: ${({ theme }) => theme.colors.black};
`;

export { Wrap, Icon, Text };
