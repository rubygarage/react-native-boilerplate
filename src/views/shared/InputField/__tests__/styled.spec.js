import React from 'react';

import renderStyled from 'utils/testHelpers/renderStyled';
import * as S from '../styled';

describe('Wrap styled components', () => {
  describe('S.Wrap matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.Wrap color />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.Label matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.Label />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.FieldWrap matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.FieldWrap />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.Field matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.Field />);
      expect(wrapper).toMatchSnapshot();
    });

    it('error', () => {
      const wrapper = renderStyled(<S.Field isError />);
      expect(wrapper).toMatchSnapshot();
    });

    it('focused', () => {
      const wrapper = renderStyled(<S.Field isFocused />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.FieldError matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.FieldError />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.FieldErrorClose matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.FieldErrorClose />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.FieldErrorMessage matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.FieldErrorMessage />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.ActionButtonWrapper matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.ActionButtonWrapper />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
