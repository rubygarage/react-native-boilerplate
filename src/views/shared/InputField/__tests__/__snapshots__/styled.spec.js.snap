// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrap styled components S.ActionButtonWrapper matches snapshot default 1`] = `
<View
  style={
    Array [
      Object {
        "height": 48,
        "justifyContent": "center",
        "position": "absolute",
        "right": 8,
      },
    ]
  }
/>
`;

exports[`Wrap styled components S.Field matches snapshot default 1`] = `
<TextInput
  allowFontScaling={true}
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "borderColor": "#000000",
        "borderRadius": 3,
        "borderStyle": "solid",
        "borderWidth": 1,
        "color": "#000000",
        "fontSize": 14,
        "lineHeight": 20,
        "minHeight": 48,
        "paddingBottom": 10,
        "paddingLeft": 12,
        "paddingRight": "NaNpx",
        "paddingTop": 10,
      },
    ]
  }
  underlineColorAndroid="transparent"
/>
`;

exports[`Wrap styled components S.Field matches snapshot error 1`] = `
<TextInput
  allowFontScaling={true}
  isError={true}
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "borderColor": "#000000",
        "borderRadius": 3,
        "borderStyle": "solid",
        "borderWidth": 1,
        "color": "#000000",
        "fontSize": 14,
        "lineHeight": 20,
        "minHeight": 48,
        "paddingBottom": 10,
        "paddingLeft": 12,
        "paddingRight": "NaNpx",
        "paddingTop": 10,
      },
    ]
  }
  underlineColorAndroid="transparent"
/>
`;

exports[`Wrap styled components S.Field matches snapshot focused 1`] = `
<TextInput
  allowFontScaling={true}
  isFocused={true}
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "borderColor": "#000000",
        "borderRadius": 3,
        "borderStyle": "solid",
        "borderWidth": 2,
        "color": "#000000",
        "fontSize": 14,
        "lineHeight": 20,
        "minHeight": 48,
        "paddingBottom": 10,
        "paddingLeft": 12,
        "paddingRight": "NaNpx",
        "paddingTop": 10,
      },
    ]
  }
  underlineColorAndroid="transparent"
/>
`;

exports[`Wrap styled components S.FieldError matches snapshot default 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 8,
        "marginTop": 6,
      },
    ]
  }
/>
`;

exports[`Wrap styled components S.FieldErrorClose matches snapshot default 1`] = `
<View
  style={
    Array [
      Object {
        "marginRight": 8,
      },
    ]
  }
/>
`;

exports[`Wrap styled components S.FieldErrorMessage matches snapshot default 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 14,
        "lineHeight": 20,
      },
    ]
  }
/>
`;

exports[`Wrap styled components S.FieldWrap matches snapshot default 1`] = `
<View
  style={
    Array [
      Object {
        "position": "relative",
      },
    ]
  }
/>
`;

exports[`Wrap styled components S.Label matches snapshot default 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 14,
        "lineHeight": 20,
        "marginBottom": 4,
      },
    ]
  }
/>
`;

exports[`Wrap styled components S.Wrap matches snapshot default 1`] = `
<View
  color={true}
  style={
    Array [
      Object {
        "flexDirection": "column",
      },
    ]
  }
/>
`;
